action.yml

```diff
@@ -35,6 +35,24 @@ runs:
       shell: bash
       run: rustup default stable
 
+    - name: Configure poetry virtualenv location
+      shell: bash
+      run: poetry config virtualenvs.in-project true
+
+    - name: Get python version
+      id: python-version
+      shell: bash
+      run: echo "version=$(python --version | sed 's/.* \([0-9]\+\.[0-9]\+\).*/\1/')" >> $GITHUB_OUTPUT
```

Comment:
My one concern here is that there's no guarantee that Poetry picks this version of Python (for instance, if `python3.8` and `python3.10` are both on the path). Maybe this isn't an issue with our docker images though?

Reply:
It's not a problem with our docker images. It would be nice if this was safe to use out side of our docker images. The problem you descibe only occurs if a python symlink exists (otherwise this action fails anyway) and that python symlink points to a different python to the one poetry was installed with, or if the project doesn't support the python that poetry was installed with. 

That second point is pretty tricky if we have two pythons installed and poetry was installed with one but the project only supports the other then poetry will I think automatically pick the other for the virtualenv. And I don't know how to check that before the virtualenv is created. 

Maybe I should make the python version an argument and then do poetry use. That way it's forward compatible to us doing matrixes of python versions.